File: server/models/Prompt.js
const mongoose = require("mongoose");

const promptSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  category_id: { type: mongoose.Schema.Types.ObjectId, ref: "Category", required: true },
  sub_category_id: { type: mongoose.Schema.Types.ObjectId, ref: "SubCategory", required: true },
  prompt: { type: String, required: true },
  response: { type: String, required: true },
  created_at: { type: Date, default: Date.now }
});

module.exports = mongoose.model("Prompt", promptSchema);

File: server/models/Prompt.py
 # models/prompt.py

from mongoengine import Document, StringField, ReferenceField, DateTimeField
from datetime import datetime
from models.User import User
from models.Category import Category
from models.SubCategory import SubCategory  # נניח שגם זה קיים

class Prompt(Document):
    user_id = ReferenceField(User, required=True)
    category_id = ReferenceField(Category, required=True)
    sub_category_id = ReferenceField(SubCategory, required=True)
    prompt = StringField(required=True)
    response = StringField(required=True)
    created_at = DateTimeField(default=datetime.utcnow)

    meta = {
        'collection': 'prompts'
    }
